onnxruntime/core/mlas/lib/power/qgemm_kernel_power10.cpp

```diff
@@ -0,0 +1,1199 @@
+                vsum[3] = RowSumBuffer[RowCount] + ColumnSumBuffer[pos+3];
+                rowC = reinterpret_cast<__vector int *>(&C[ldc * RowCount + VectorCount]);
+                rowC[0] = *reinterpret_cast<__vector int *>(&result[RowCount]) + vsum;
+            }
+        } else {
+            for (size_t RowCount = 0;RowCount < row; RowCount++){
+                vsum[0] = RowSumBuffer[RowCount] + ColumnSumBuffer[pos];
+                vsum[1] = RowSumBuffer[RowCount] + ColumnSumBuffer[pos+1];
+                vsum[2] = RowSumBuffer[RowCount] + ColumnSumBuffer[pos+2];
+                vsum[3] = RowSumBuffer[RowCount] + ColumnSumBuffer[pos+3];
+                rowC = reinterpret_cast<__vector int *>(&C[ldc * RowCount + VectorCount]);
+                rowC[0] += *reinterpret_cast<__vector int *>(&result[RowCount]) + vsum;
+            }
+        }
+    }
+};
+template<size_t Lane>
+MLAS_FORCEINLINE
+void
+MlasQgemmStoreScalarMMA(
+    MLAS_INT32X4 result[4],
+    int32_t* C,
+    size_t ldc,
+    size_t row,
+    bool ZeroMode,
+    const int32_t* RowSumBuffer,
+    const int32_t* ColumnSumBuffer,
+    const int32_t* ZeroPointB
+    )
+{
+    if (ZeroPointB != nullptr) {
+        if (ZeroMode) {
+            for (size_t RowCount = 0;RowCount < row; RowCount++){
+                int sum = RowSumBuffer[RowCount];
+                sum *= ZeroPointB[0];
+                sum += ColumnSumBuffer[0];
+                C[RowCount*ldc+Lane] = result[RowCount][Lane] + sum;
+            }
+        } else {
+            for (size_t RowCount = 0;RowCount < row; RowCount++){
+                int sum = RowSumBuffer[RowCount];
+                sum *= ZeroPointB[0];
+                sum += ColumnSumBuffer[0];
+                C[RowCount*ldc+Lane] += result[RowCount][Lane] + sum;
+            }
+        }
+    } else {
+        if (ZeroMode) {
+            for (size_t RowCount = 0;RowCount < row; RowCount++){
+                int sum = RowSumBuffer[RowCount] + ColumnSumBuffer[0];
+                C[RowCount*ldc+Lane] = result[RowCount][Lane] + sum;
+            }
+        } else {
+            for (size_t RowCount = 0;RowCount < row; RowCount++){
+                int sum = RowSumBuffer[RowCount] + ColumnSumBuffer[0];
+                C[RowCount*ldc+Lane] += result[RowCount][Lane] + sum;
+            }
+        }
+    }
+};
+template<>
+size_t
+MlasGemmQuantKernel<MLAS_GEMM_QUANT_KERNEL_POWER10>(
+    const MLAS_GEMM_QUANT_KERNEL_POWER10::PackedAType* A,
+    const MLAS_GEMM_QUANT_KERNEL_POWER10::PackedBType* B,
+    int32_t* C,
+    size_t PackedCountK,
+    size_t CountM,
+    size_t CountN,
+    size_t ldc,
+    const int32_t* RowSumBuffer,
+    const int32_t* ColumnSumBuffer,
+    const int32_t* ZeroPointB,
+    bool ZeroMode
+    )
+{
+    MLAS_UNREFERENCED_PARAMETER(CountM);
+    MLAS_UNREFERENCED_PARAMETER(ldc);
+    if (CountM < 8 && CountM >= 4) {
+        CountM = 4;
+    }
+    size_t Mval = CountM;
```

Comment:
>Mval [](http://example.com/codeflow?start=11&length=4)

nit: m

Reply:
variable 'Mval' is used to iterate the number of rows in storing results from the accumulator. For cases where CountM is greater than 8, this has to be restricted to 4. So we need a different variable other than CountM.